packages/nextjs/components/ExampleUi/ContractInteraction/PurposeSetter.tsx

```diff
@@ -10,11 +27,31 @@ export const PurposeSetter = () => {
           type="text"
           placeholder="Write your purpose here"
           className="input font-bai-jamjuree w-full px-5 bg-[url('/assets/gradient-bg.png')] bg-[length:100%_100%] border border-primary text-lg sm:text-2xl placeholder-white uppercase"
+          onChange={e => setNewPurpose(e.target.value)}
         />
         <div className="flex rounded-full border border-primary p-1 flex-shrink-0">
           <div className="flex rounded-full border-2 border-primary p-1">
-            <button className="btn btn-primary rounded-full capitalize font-normal font-white w-24 flex items-center gap-1 hover:gap-2 transition-all tracking-widest">
-              Send <ArrowSmallRightIcon className="w-3 h-3 mt-0.5" />
+            <button
+              className={`btn btn-primary rounded-full capitalize font-normal font-white w-24 flex items-center gap-1 hover:gap-2 transition-all tracking-widest ${
+                isLoading ? "loading" : ""
+              }`}
+              onClick={async () => {
+                if (writeAsync && writeTxn) {
+                  try {
+                    // If the contract is not deployed this "works". I get a "mined successfully" msg.
+                    await writeTxn(writeAsync());
```

Comment:
If the contract is not deployed this "works". I get a "mined successfully" msg.

Maybe we are not handling well errors in the write tx

Reply:
Ohh yes !! this shouldn't happen, Looking into it 🙌

Reply:
Created https://github.com/scaffold-eth/se-2/issues/129